airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -180,8 +185,12 @@ private static boolean shouldUseCDC(final ConfiguredAirbyteCatalog catalog) {
           new AirbyteDebeziumHandler(sourceConfig, MySqlCdcTargetPosition.targetPosition(database), MySqlCdcProperties.getDebeziumProperties(),
               catalog, true);
 
-      return handler.getIncrementalIterators(new MySqlCdcSavedInfoFetcher(stateManager.getCdcStateManager().getCdcState()),
-          new MySqlCdcStateHandler(stateManager), new MySqlCdcConnectorMetadataInjector(), emittedAt);
+      CdcState cdcState = stateManager.getCdcStateManager().getCdcState();
```

Comment:
Could `stateManager.getCdcStateManager()` return null?

Reply:
no, from what I can see from the code :
 public StateManager(final DbState serialized, final ConfiguredAirbyteCatalog catalog) {
    this.cdcStateManager = new CdcStateManager(serialized.getCdcState());
    this.isCdc = serialized.getCdc();
    if (serialized.getCdc() == null) {
      this.isCdc = false;
    }